tests/test_flows_seed.py

```diff
@@ -89,42 +91,181 @@ def test_invalid_mnemonic(self):
+        self.settings.set_value(SettingsConstants.SETTING__SCRIPT_TYPES, [x for x,y in script_types])
+        self.settings.set_value(SettingsConstants.SETTING__COORDINATORS, [x for x,y in coordinators])
+
+        # exhaustively test flows thru standard sig_types, script_types, and coordinators
+        for sig_tuple in sig_types:
+            for script_tuple in script_types:
+                for coord_tuple in coordinators:
+                    # skip custom derivation
+                    if script_tuple[0] == SettingsConstants.CUSTOM_DERIVATION:
+                        continue 
+                    # skip multisig taproot
+                    elif sig_tuple[0] == SettingsConstants.MULTISIG and script_tuple[0] == SettingsConstants.TAPROOT:
+                        continue
+                    else:
+                        print('\n\ntest_standard_xpubs(%s, %s, %s)' % (sig_tuple, script_tuple, coord_tuple))
+                        test_standard_xpubs(sig_tuple, script_tuple, coord_tuple)
+
+
+    def test_export_xpub_disabled_not_available_flow(self):
+        """
+            If sig_type/script_type/coordinator disabled, then these options are not available
+        """
+        # Load a finalized Seed into the Controller
+        mnemonic = "blush twice taste dawn feed second opinion lazy thumb play neglect impact".split()
+        self.controller.storage.set_pending_seed(Seed(mnemonic=mnemonic))
+        self.controller.storage.finalize_pending_seed()
+
+        # these are (constant_value, display_name) tuples
+        sig_types = SettingsConstants.ALL_SIG_TYPES
+        script_types = SettingsConstants.ALL_SCRIPT_TYPES
+        coordinators = SettingsConstants.ALL_COORDINATORS
+
+        # these are the disabled types that we will be testing
+        disabled_sig = SettingsConstants.MULTISIG
+        disabled_script = SettingsConstants.TAPROOT
+        disabled_coord = SettingsConstants.COORDINATOR__NUNCHUK
+
+        # enable all but our target disabled type
+        self.settings.set_value(SettingsConstants.SETTING__SIG_TYPES, [x for x,y in sig_types if x!=disabled_sig])
+        self.settings.set_value(SettingsConstants.SETTING__SCRIPT_TYPES, [x for x,y in script_types if x!=disabled_script])
+        self.settings.set_value(SettingsConstants.SETTING__COORDINATORS, [x for x,y in coordinators if x!=disabled_coord])
+
+        # test that multisig is not an option via exception raised when NOT redirected to next step w/o a choice
+        with pytest.raises(FlowBasedTestUnexpectedViewError) as e:
+            self.run_sequence(
+                initial_destination_view_args=dict(seed_num=0),
+                sequence=[
+                    FlowStep(seed_views.SeedOptionsView, button_data_selection=seed_views.SeedOptionsView.EXPORT_XPUB),
+                    FlowStep(seed_views.SeedExportXpubSigTypeView, button_data_selection=disabled_sig),
```

Comment:
Oof, there's a subtle bug here. With multisig disabled, the only remaining sig type is single sig. Therefore that View should automatically redirect past this View and on to the next View (script type selection), which it does. But the test flow doesn't notice the expectation mismatch. The entry should be:
```
FlowStep(seed_views.SeedExportXpubSigTypeView, is_redirect=True)
```

I'll have to look at if I can do more rigorous validation on the button_data_selection; this sequence should fail if it's written to expect to be able to select a given button but the View redirects away without a user selection.

Reply:
Darn.  testing is kinda hard...  it's not always that a test is a "good" test.  I was trying to prove that an exception would be raised, because indeed it should redirect when only one choice, which it does... but I was trying to fake the flow test into having a choice "disabled_sig" so it would fail and raise an Exception.  But it's not so expressive and clear of a test.